### PR TITLE
Better fix for TerminalReporter issue

### PR DESCRIPTION
**Pull request by BeyondEvil**
The [issue](https://github.com/pytest-dev/pytest/issues/6936) with TerminalWriter in `pytester` is causing tests to fail.

### CODE REVIEW

File: testing/test_pytest_html.py [hunk 1]
```diff
@@ -14,20 +14,15 @@
...
+    # TODO: Temporary hack until they fix
     # https://github.com/pytest-dev/pytest/issues/6936
-    import warnings
-    from _pytest.warnings import _setoption
-
-    arg = "ignore:TerminalReporter.writer:pytest.PytestDeprecationWarning"
-    _setoption(warnings, arg)
+    for index, item in enumerate(recwarn):
+        if "TerminalReporter.writer" in repr(item.message):
+            del recwarn.list[index]
```
Comment (dhalperi) [inline testing/test_pytest_html.py:22]: I don't think this is really safe if there are multiple instances in the list, but in practice that likely does not matter here.

(if you delete index 0 in a list of length 2, enumerate will never even product the second item).

Comment (BeyondEvil) [reply testing/test_pytest_html.py:22]: Aha, is it better to store the relevant indexes and then after the iteration is complete delete them? @dhalperi

Comment (dhalperi) [inline testing/test_pytest_html.py:22]: I might do something like this: (sorry for laziness, couldn't get github multiline suggest to work)

```py
def filter_warnings(recwarn):
  # TODO..
  return [w for w in recwarn.list if "TerminalReporter.writer" in repr(w.message)]
```

and then below

```py
warnings = filter_warnings(recwarn)
assert len(warnings) == 0
```

Comment (dhalperi) [reply testing/test_pytest_html.py:22]: see suggestion - basically, for me, not modifying recwarn seems easier than tracking indices, etc.

Comment (BeyondEvil) [reply testing/test_pytest_html.py]: Thanks! 🙏

**Comment by BeyondEvil** [comment]
Seems like Travis is having some issues. Will try again tomorrow.

**Comment by BeyondEvil** [comment]
@nicoddemus Sorry for spamming with reviewers, but this is blocking other PRs.

File: testing/test_pytest_html.py [hunk 2]
```diff
@@ -894,9 +887,6 @@ def test_css_invalid(self, testdir, recwarn):
         assert "No such file or directory: 'style.css'" in result.stderr.str()
 
     def test_css_invalid_no_html(self, testdir):
```
Comment (nicoddemus) [inline testing/test_pytest_html.py:889]: If you mark the tests to filter warnings:

```python
@pytest.mark.filterwarnings("ignore:.*TerminalReporter.writer.*")
```

You might not need `remove_deprecation_from_recwarn` in the end?

**Comment by nicoddemus** [comment]
> @nicoddemus Sorry for spamming with reviewers, but this is blocking other PRs.

No worries, I appreciate the ping. 👍

File: testing/test_pytest_html.py (hunk 2 above)
Comment (BeyondEvil) [reply testing/test_pytest_html.py:889]: Tried it and it didn't work @nicoddemus 

Maybe I'm doing something wrong?

```python
    @pytest.mark.filterwarnings("ignore:.*TerminalReporter.writer.*")
    @pytest.mark.parametrize("colors", [(["red"]), (["green", "blue"])])
    def test_css(self, testdir, recwarn, colors):
        testdir.makepyfile("def test_pass(): pass")
        css = {}
        cssargs = []
        for color in colors:
            style = f"* {{color: {color}}}"
            path = testdir.makefile(".css", **{color: style})
            css[color] = {"style": style, "path": path}
            cssargs.extend(["--css", path])
        result, html = run(testdir, "report.html", "--self-contained-html", *cssargs)
        assert result.ret == 0
        assert len(recwarn) == 0
        for k, v in css.items():
            assert str(v["path"]) in html
            assert v["style"] in html
```

Comment (nicoddemus) [reply testing/test_pytest_html.py:889]: Yeah unfortunately the mark is considered only for the test run, not the setup (which is where the warning is issued).

**Pull request merged**